File: 1087-longest-arithmetic-subsequence/longest-arithmetic-subsequence.cpp
class Solution {
public:
    int longestArithSeqLength(vector<int>& nums) {
        int n=nums.size();
        if(n<=2)
            return n;

        // //Step 1 : Create Map
        unordered_map<int,int>dp[n+1];

        int ans=0;
        for(int i=0;i<n;i++){
            for(int j=0;j<i;j++){
                int diff=nums[i]-nums[j];
                int count=1;

                //Checking if answer already present
                if(dp[j].count(diff)){
                    //Put it in count
                    count = dp[j][diff];
                }
                dp[i][diff]=1+count;  // 1----> Current element ko include karna hai
                ans=max(ans,dp[i][diff]);
            }
        }
        return ans;
    }
};

File: 1087-longest-arithmetic-subsequence/longest-arithmetic-subsequence.py
class Solution:
    def longestArithSeqLength(self, nums: List[int]) -> int:
        ans = 0
        n = len(nums)
        dp = [defaultdict(int) for _ in range(n)]
        for i in range(n):
            for j in range(i):
                diff = nums[j] - nums[i]
                dp[i][diff] = dp[j].get(diff, 1) + 1
                ans = max(ans, dp[i][diff])
        return ans